web/app/Http/Controllers/JobsController.php

```diff
@@ -17,9 +18,23 @@ class JobsController extends Controller
      *
      * @return \Illuminate\Http\Response
      */
-    public function index()
+    public function index(Request $request)
     {
-      $jobs = Jobs::orderBy('created_at', 'desc')->get();
+      $jobs = [];
+      $type = $request->get('type');
+      if($type && $type === 'need_approval') {
+          if(Auth::user()->hasUserRole && Auth::user()->hasUserRole->hasRole->name === 'moderator')
```

Comment:
Define `Auth::user()` as `$user` to avoid multiple calls to the `Auth::user()` method